nucliadb/nucliadb/ingest/orm/resource.py

```diff
@@ -606,183 +610,198 @@ async def apply_extracted(self, message: BrokerMessage):
+        maybe_update_basic_thumbnail(self.basic, link_extracted_data.link_thumbnail)
+
+        await field_link.set_link_extracted_data(link_extracted_data)
+
+        if self.basic.icon == "":
+            self.basic.icon = "application/stf-link"
+
+        if (
+            self.basic.title.startswith("http") and link_extracted_data.title != ""
+        ) or (self.basic.title == "" and link_extracted_data.title != ""):
+            # If the title was http something or empty replace
+            self.basic.title = link_extracted_data.title
+
+        maybe_update_basic_summary(self.basic, link_extracted_data.description)
+
+    async def _apply_file_extracted_data(self, file_extracted_data: FileExtractedData):
+        assert self.basic is not None
+        field_file: File = await self.get_field(
+            file_extracted_data.field,
+            FieldType.FILE,
+            load=False,
+        )
+        if file_extracted_data.icon != "" and (
+            self.basic.icon == "" or self.basic.icon == "application/octet-stream"
+        ):
+            self.basic.icon = file_extracted_data.icon
+
+        maybe_update_basic_thumbnail(self.basic, file_extracted_data.file_thumbnail)
+
+        await field_file.set_file_extracted_data(file_extracted_data)
+
+    async def _apply_field_computed_metadata(
+        self, field_metadata: FieldComputedMetadataWrapper
+    ):
+        assert self.basic is not None
+        maybe_update_basic_summary(self.basic, field_metadata.metadata.metadata.summary)
+
+        field_obj = await self.get_field(
+            field_metadata.field.field,
+            field_metadata.field.field_type,
+            load=False,
+        )
+        (
+            metadata,
+            replace_field,
+            replace_splits,
+        ) = await field_obj.set_field_metadata(field_metadata)
+        field_key = self.generate_field_id(field_metadata.field)
+
+        page_positions: Optional[FilePagePositions] = None
+        if field_metadata.field.field_type == FieldType.FILE and isinstance(
+            field_obj, File
+        ):
+            page_positions = await get_file_page_positions(field_obj)
+
+        user_field_metadata = next(
+            (
+                fm
+                for fm in self.basic.fieldmetadata
+                if fm.field.field == field_metadata.field.field
+                and fm.field.field_type == field_metadata.field.field_type
+            ),
+            None,
+        )
+
+        apply_field_metadata = partial(
+            self.indexer.apply_field_metadata,
+            field_key,
+            metadata,
+            replace_field=replace_field,
+            replace_splits=replace_splits,
+            page_positions=page_positions,
+            extracted_text=await field_obj.get_extracted_text(),
+            basic_user_field_metadata=user_field_metadata,
+        )
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(None, apply_field_metadata)
```

Comment:
What do you think about creating an executor we control for this?

So maybe just create an executor at the module level:

```python
_executor = ThreadPoolExecutor(10)
```

I feel like we want to use more threads and optimize for more even performance with concurrency over individual operation performance.

Reply:
Good idea

Reply:
I've added `apply_field_vectors` to also run in an executor